### ISSUE TITLE
No math support at all?

### ISSUE THREAD

**Issue by bk322** [marianoguerra/rst2html5#42]
`rst2html` supports math:

```
.. math::

   x^2
```

divs get some math processing depending on the command line options of the `rst2html`. However `rst2html5` just leaves plain divs.

**Comment by torfsen** [comment]
It seems that non-MathJax math support was removed in 2b667c46e38dc285a7953f1a19a1195018af8e7d when a new Writer was introduced. It looks like MathML and other options were supported before that. I'd also prefer to have more choices than just MathJax, and, more important, I'd like to use RST markup for declaring math (i.e. the `:math:` role and the `.. math::` directive). With the current MathJax support, on the other hand, it seems that I need to use MathJax markup in my RST files. That's rather contrary to the idea of keeping semantics (RST) and display (HTML, CSS, MathJax) separate.

In my opinion, the ideal solution would be to have an option for selecting the math output variant (MathML, MathJax, etc.) and the `:math:` role and `.. math::` directive would then be translated into appropriate code. That would allow users to switch from MathML to MathJax (and vice versa) by simply changing the options they use to call `rst2html5`. If I understand things correctly that is also the approach that was used in the old writer.

@marianoguerra: Is that something that would be OK for you? If so I would try to implement that and create a PR.

**Comment by marianoguerra** [comment]
hi, thanks for the comment, I don't use math myself so you seem to know more than I do, please create a PR and I will merge it, I guess it's better than the current state of things :)

**Comment by torfsen** [comment]
I'll prepare PR, although it may take me time since things are pretty busy here at the moment.

**Comment by torfsen** [comment]
I'm working on this. Most of it is done, but some small issues still need to be fixed. Basically I took the code from the original docutils, refactored it and adapted it to our architecture.

@marianoguerra: Once everything works I would remove the current MathJax-postprocessor, is that OK with you?

**Comment by marianoguerra** [comment]
@torfsen if the replacement works, I'm ok with it.

**Comment by torfsen** [comment]
This should be ready to go. Documentation (well ,the README) and tests have been updated.

I've added new configuration options (`--mathjax-url`, `--mathjax-config`, `--math-css`) to get away from previous means (`--math-output "mathjax foo"` and `--mathjax-opts url=foo,config=bar`) which I find not very user friendly (and harder to handle in the code, too). The old approaches are still supported (but not documented), though.

I've also decided to keep the MathJax postprocessor to make sure that we stay backwards-compatible. However I removed its detailed documentation and marked it as deprecated instead.
